### PR TITLE
2586 ic switch   focus state is not visible in high contrast mode

### PR DESCRIPTION
**Pull request by cd3859**
## Summary of the changes
Added css for high contrast focus of the ic-switch

## Related issue
#2586 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

**Review by ad9242 — DISMISSED** [review]

File: packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png
Comment (GCHQ-Developer-299) [inline packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png:1]: are other focus indicators in HCM this subtle? I'll need to compare.

**Review by GCHQ-Developer-299 — CHANGES_REQUESTED** [review]
Can you make the focussed state less subtle please?  As a comparison, below is a screenshot of how buttons look when focussed in HCM

Screenshot 2024-10-24 at 16 00 38

File: packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png
Comment (cd3859) [reply packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png:1]: I've used the dismissible chips as reference, the issue with a more bold border line is it changes the shape of the switch

Comment (ad9242) [reply packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png:1]: yeah, I've had a quick look too, but as @cd3859 says, increasing the border size causes movement of the inner content when focus is added\removed, so looks a bit jarring.

Comment (MI6-286) [reply packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png:1]: On a Quick Look through the VR screenshots we have for focus in HCM, it seems there are a range of thicknesses. Might be worth a wider ticket to look into it.

Comment (ad9242) [reply packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png:1]: maybe the solution is to go with the "outline" fix you had previously, as that seems to allow you to increase the width, without affecting the inner content. the visual regression base-line will need updating again too - sorry!

Comment (cd3859) [reply packages/react/cypress-image-diff-screenshots/baseline/IcSwitch.cy.tsx-focus-high-contrast.png:1]: @GCHQ-Developer-299 added outline similar to checkbox and radio buttons to emphasize focus

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**